fastlane/lib/fastlane/actions/device_grid/README.md

```diff
@@ -0,0 +1,106 @@
+# fastlane danger Device Grid
+
+Ever dream of testing your app straight from a pull request? Well now you can! With [fastlane](https://fastlane.tools), [danger](https://github.com/danger/danger) and [appetize.io](https://appetize.io/), you can stream your latest changes right from the browser.
+
+No more manually installing and testing your app just to review a PR.
+
+![assets/GridExampleScreenshot.png](assets/GridExampleScreenshot.png)
+
+## Requirements
+
+- [fastlane](https://fastlane.tools)
+- [danger](https://github.com/danger/danger)
+- [appetize.io](https://appetize.io/) account
+- A Continuous Integration system
+
+## Getting started
+
+### Install fastlane and danger
+
+Create a `Gemfile` in your project's directory with the following content
+
+```ruby
+gem "fastlane"
+gem "danger"
+```
+
+and run
+
+```
+bundle install
+```
+
+### Setup `fastlane`
+
+Skip this step if you're already using `fastlane` (which you should)
+
+```
+fastlane init
+```
+
+### Setup `danger`
+
+```
+danger init
+```
+
+Follow the `danger` guide to authenticate with GitHub
+
+### Configure [fastlane](https://fastlane.tools)
+
+Edit `fastlane/Fastfile`. Feel free to remove the auto-generated lanes. Add the following lane:
+
+```ruby
+desc "Build your app and upload it to Appetize to stream it in your browser"
+lane :upload_to_appetize do
+  import_from_git(url: "https://github.com/fastlane/fastlane",
+                 path: "fastlane/lib/fastlane/actions/device_grid/AppetizeFastfile")
```

Comment:
I'm wondering if we'll get into trouble here, with this part being pulled directly from git and it's dependencies being versioned like RubyGems.  E.g. if we want to change behavior based on `gym` or `fastlane` updates.